core/state/statedb.go

```diff
@@ -1102,3 +1125,13 @@ func (s *StateDB) AddressInAccessList(addr common.Address) bool {
 func (s *StateDB) SlotInAccessList(addr common.Address, slot common.Hash) (addressPresent bool, slotPresent bool) {
 	return s.accessList.Contains(addr, slot)
 }
+
+// GetPredicateStorageSlots returns whether or not the given address was included in the optional
+// access list of the transaction, and the storage slots that were associated with that address
+// if it was included. The storage slots are returned in the same order as they appeared in the transaction.
+// These are the same storage slots that are used to verify any transaction
+// predicates for transactions with access list addresses that match a precompile address.
+func (s *StateDB) GetPredicateStorageSlots(address common.Address) (bool, []common.Hash) {
+	storageSlots, exists := s.predicateStorageSlots[address]
+	return exists, storageSlots
```

Comment:
i don't think we should extract these from the map and then return them in reverse order